app/repositories/slick/implementations/SlickChatsRepository.scala

```diff
@@ -93,57 +93,75 @@ class SlickChatsRepository @Inject() (db: Database)(implicit executionContext: E
     } yield (chatId, emailId, body, date, sent)
   //endregion
 
+  def cleanRepeatedDateEmails(chatPreviewSeq: Seq[ChatPreview]): Seq[ChatPreview] = {
+    val indexed = chatPreviewSeq.zipWithIndex
+    indexed.filterNot {
+      case (chatprev, idx) if idx > 0 =>
+        (chatprev.lastEmailDate, chatprev.chatId) == (indexed(idx - 1)._1.lastEmailDate, indexed(idx - 1)._1.chatId)
+      case (chatprev, idx) if idx == 0 => false
+    }.map(_._1)
+  }
+
   /**
-   * Method that returns a preview of all the chats of a given user in a given Mailbox
+   * Creates a DBIOAction that returns a preview of all the chats of a given user in a given Mailbox
    * @param mailbox The mailbox being seen
    * @param userId The userId of the user in question
-   * @return A Future sequence of ChatPreview dtos. The preview of each chat only shows the most recent email
+   * @return A DBIOAction that when run returns a sequence of ChatPreview dtos.
+   *         The preview of each chat only shows the most recent email
    */
-  def getChatsPreview(mailbox: Mailbox, userId: String): Future[Seq[ChatPreview]] = {
+  def getChatsPreviewAction(mailbox: Mailbox, userId: String) = {
 
-    val groupedQuery = getVisibleEmailsQuery(userId, Some(mailbox))
+    val visibleEmailsQuery = getVisibleEmailsQuery(userId, Some(mailbox))
+
+    val groupedQuery = visibleEmailsQuery
       .map(visibleEmailRow => (visibleEmailRow._1, visibleEmailRow._4))
       .groupBy(_._1)
       .map { case (chatId, date) => (chatId, date.map(_._2).max) }
 
     val chatPreviewQuery = for {
-      (chatId, date) <- groupedQuery
+      (chatId, maxDate) <- groupedQuery
       subject <- ChatsTable.all.filter(_.chatId === chatId).map(_.subject)
-      (emailId, body) <- EmailsTable.all.filter(emailRow =>
-        emailRow.chatId === chatId && emailRow.date === date).map(emailRow =>
-        (emailRow.emailId, emailRow.body.take(PREVIEW_BODY_LENGTH)))
+      (emailId, body, date) <- EmailsTable.all.filter(emailRow =>
+        emailRow.chatId === chatId && emailRow.date === maxDate).map(emailRow =>
```

Comment:
Don't see the point of using dates to filter something. It smells bad.

Reply:
I know, but I have no choice. 
Like we talked before the previous groupBy only allows me to take the chatId and date columns. 
So I am forced to use the date to make a join that gives me the information I require.